tests/codegen/issues/issue-111603.rs

```diff
@@ -5,6 +5,18 @@
 
 use std::sync::Arc;
 
+// CHECK-LABEL: @new_from_array
```

Comment:
Could we emit the entire IR here? I think its helpful for reviewers, test coverage, and people trying to understand the codebase better.

Reply:
Because we don't have something like LLVM's update_test_checks script, it is painful to update codegen tests. So we tend to use the minimal amount of CHECK lines so we don't have to update tests due to unrelated changes. (Also, most tests don't specify a target, so they generate slightly different IR per platform due to ABI differences etc., which would have to be dealt with.)

In this case, this is the full IR of `new_from_array`:

<details>
<summary>IR</summary>

```ll
define { ptr, i64 } @new_from_array(i64 noundef %x) unnamed_addr #0 personality ptr @rust_eh_personality {
start:
  %array = alloca [1000 x i64], align 8
  %broadcast.splatinsert = insertelement <2 x i64> poison, i64 %x, i64 0
  %broadcast.splat = shufflevector <2 x i64> %broadcast.splatinsert, <2 x i64> poison, <2 x i32> zeroinitializer
  %broadcast.splatinsert1 = insertelement <2 x i64> poison, i64 %x, i64 0
  %broadcast.splat2 = shufflevector <2 x i64> %broadcast.splatinsert1, <2 x i64> poison, <2 x i32> zeroinitializer
  br label %vector.body

vector.body:                                      ; preds = %vector.body, %start
  %index = phi i64 [ 0, %start ], [ %index.next.4, %vector.body ]
  %0 = getelementptr inbounds [1000 x i64], ptr %array, i64 0, i64 %index
  store <2 x i64> %broadcast.splat, ptr %0, align 8
  %1 = getelementptr inbounds i64, ptr %0, i64 2
  store <2 x i64> %broadcast.splat2, ptr %1, align 8
  %index.next = add nuw nsw i64 %index, 4
  %2 = getelementptr inbounds [1000 x i64], ptr %array, i64 0, i64 %index.next
  store <2 x i64> %broadcast.splat, ptr %2, align 8
  %3 = getelementptr inbounds i64, ptr %2, i64 2
  store <2 x i64> %broadcast.splat2, ptr %3, align 8
  %index.next.1 = add nuw nsw i64 %index, 8
  %4 = getelementptr inbounds [1000 x i64], ptr %array, i64 0, i64 %index.next.1
  store <2 x i64> %broadcast.splat, ptr %4, align 8
  %5 = getelementptr inbounds i64, ptr %4, i64 2
  store <2 x i64> %broadcast.splat2, ptr %5, align 8
  %index.next.2 = add nuw nsw i64 %index, 12
  %6 = getelementptr inbounds [1000 x i64], ptr %array, i64 0, i64 %index.next.2
  store <2 x i64> %broadcast.splat, ptr %6, align 8
  %7 = getelementptr inbounds i64, ptr %6, i64 2
  store <2 x i64> %broadcast.splat2, ptr %7, align 8
  %index.next.3 = add nuw nsw i64 %index, 16
  %8 = getelementptr inbounds [1000 x i64], ptr %array, i64 0, i64 %index.next.3
  store <2 x i64> %broadcast.splat, ptr %8, align 8
  %9 = getelementptr inbounds i64, ptr %8, i64 2
  store <2 x i64> %broadcast.splat2, ptr %9, align 8
  %index.next.4 = add nuw nsw i64 %index, 20
  %10 = icmp eq i64 %index.next.4, 1000
  br i1 %10, label %repeat_loop_next, label %vector.body, !llvm.loop !2

repeat_loop_next:                                 ; preds = %vector.body
  %11 = load volatile i8, ptr @__rust_no_alloc_shim_is_unstable, align 1, !noalias !5
  %12 = tail call noundef align 8 dereferenceable_or_null(8016) ptr @__rust_alloc(i64 noundef 8016, i64 noundef 8) #6, !noalias !5
  %13 = icmp eq ptr %12, null
  br i1 %13, label %bb1.i.i, label %"_ZN5alloc4sync12Arc$LT$T$GT$3new17hc22c917a7edefd8bE.exit"

bb1.i.i:                                          ; preds = %repeat_loop_next
; call alloc::alloc::handle_alloc_error
  tail call void @_ZN5alloc5alloc18handle_alloc_error17h5a822ff2e844764dE(i64 noundef 8, i64 noundef 8016) #7, !noalias !5
  unreachable

"_ZN5alloc4sync12Arc$LT$T$GT$3new17hc22c917a7edefd8bE.exit": ; preds = %repeat_loop_next
  store i64 1, ptr %12, align 8, !noalias !5
  %x.sroa.4.0._14.sroa_idx.i = getelementptr inbounds i8, ptr %12, i64 8
  store i64 1, ptr %x.sroa.4.0._14.sroa_idx.i, align 8, !noalias !5
  %x.sroa.5.0._14.sroa_idx.i = getelementptr inbounds i8, ptr %12, i64 16
  call void @llvm.memcpy.p0.p0.i64(ptr noundef nonnull align 8 dereferenceable(8000) %x.sroa.5.0._14.sroa_idx.i, ptr noundef nonnull align 8 dereferenceable(8000) %array, i64 8000, i1 false)
  %14 = insertvalue { ptr, i64 } poison, ptr %12, 0
  %15 = insertvalue { ptr, i64 } %14, i64 1000, 1
  ret { ptr, i64 } %15
}
```

</details>

Reply:
Thanks for sharing IR. Looks like the redundant alloca was removed. Nice!